Bottom: f254491a86f0f0ff170a2484744080a0ffe7733b
Top:    3936d3c488cff2085c02e1d3b96c15caa882a9ac
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2013-01-03 22:56:50 +0000

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index d06030d..bcb9a52 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
